File: mobile/src/main/java/treehou/se/habit/ui/control/cells/config/cells/DefaultConfigCellBuilder.java
package treehou.se.habit.ui.control.cells.config.cells;

import android.content.Context;
import android.graphics.PorterDuff;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageButton;
import android.widget.RemoteViews;

import butterknife.BindView;
import butterknife.ButterKnife;
import treehou.se.habit.R;
import treehou.se.habit.core.db.model.controller.CellDB;
import treehou.se.habit.core.db.model.controller.ControllerDB;
import treehou.se.habit.ui.control.CellFactory;
import treehou.se.habit.ui.control.ControllerUtil;

public class DefaultConfigCellBuilder implements CellFactory.CellBuilder {

    @BindView(R.id.img_icon_button) ImageButton imgView;

    public View build(Context context, ControllerDB controller, CellDB cell){

        LayoutInflater inflater = LayoutInflater.from(context);
        View rootView = inflater.inflate(R.layout.cell_conf_button, null);
        ButterKnife.bind(this, rootView);

        int[] pallete = ControllerUtil.generateColor(controller, cell);

        imgView.getBackground().setColorFilter(pallete[ControllerUtil.INDEX_BUTTON], PorterDuff.Mode.MULTIPLY);

        return rootView;
    }

    @Override
    public RemoteViews buildRemote(Context context, ControllerDB controller, CellDB cell) {
        return null;
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/control/cells/config/cells/DefaultConfigCellBuilder.kt
package treehou.se.habit.ui.control.cells.config.cells

import android.content.Context
import android.graphics.PorterDuff
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.RemoteViews
import treehou.se.habit.R
import treehou.se.habit.core.db.model.controller.CellDB
import treehou.se.habit.core.db.model.controller.ControllerDB
import treehou.se.habit.ui.control.CellFactory
import treehou.se.habit.ui.control.ControllerUtil

class DefaultConfigCellBuilder : CellFactory.CellBuilder {

    override fun build(context: Context, container: ViewGroup, controller: ControllerDB, cell: CellDB): View {

        val inflater = LayoutInflater.from(context)
        val rootView = inflater.inflate(R.layout.cell_conf_button, container, false)

        val pallete = ControllerUtil.generateColor(controller, cell)

        val iconButton = rootView.findViewById<ImageButton>(R.id.iconButton)
        iconButton.background.setColorFilter(pallete[ControllerUtil.INDEX_BUTTON], PorterDuff.Mode.MULTIPLY)

        return rootView
    }

    override fun buildRemote(context: Context, controller: ControllerDB, cell: CellDB): RemoteViews? {
        return null
    }
}
